server/modules/analysis/service/rChain/rFile/system/rFileReadData.js

```diff
@@ -65,33 +59,52 @@ export default class RFileReadData extends RFileSystem {
 
   async appendContentToConvertDataTypes({ entityDef }) {
     const { survey } = this.rChain
-    const contentConvertDataTypes = []
-
-    Survey.visitAncestorsAndSelf(entityDef, (ancestorDef) => {
-      Survey.getNodeDefChildren(
-        ancestorDef,
-        false
-      )(survey).forEach((childDef) => {
-        if (NodeDef.isSingleAttribute(childDef)) {
-          contentConvertDataTypes.push(...this.createContentToConvertNodeDefColumnsDataTypes({ entityDef, childDef }))
+
+    const viewDataNodeDef = new ViewDataNodeDef(survey, entityDef)
+
+    const columnsToConvertAsCharacter = []
+    const columnsToConvertAsLogical = []
+    const columnsToConvertAsNumeric = []
+
+    viewDataNodeDef.columnNodeDefs.map((columnNodeDef) => {
+      const { nodeDef, names: columnNames } = columnNodeDef
+      columnNames.forEach((colName) => {
+        if (NodeDef.isSingleAttribute(nodeDef) && !NodeDef.isAnalysis(nodeDef)) {
+          const type = NodeDef.getType(nodeDef)
+          if (NodeDef.isBoolean(nodeDef)) {
+            columnsToConvertAsLogical.push(colName)
+          } else if (
+            [
```

Comment:
It was maybe clearer before, but now it is hard to follow and maintain 

Maybe this link can help to improve this code -> https://levelup.gitconnected.com/learn-these-9-rules-to-start-writing-clean-code-immediately-a7ee40fe1e1b

Reply:
These are the sonar reports about complexity before and after
<img width="428" alt="Captura de Pantalla 2022-10-10 a las 13 28 01" src="https://user-images.githubusercontent.com/11272068/194856175-8fd6216a-2aa0-499f-8ce6-5fe69c4b00d6.png">
<img width="436" alt="Captura de Pantalla 2022-10-10 a las 13 28 09" src="https://user-images.githubusercontent.com/11272068/194856197-d73a58f5-fc10-4007-a41a-6805221ffe87.png">

before was 3-13
now is 14-18